File: mediator/java/MediatorTest.java
public class MediatorTest {
    public static void main(String[] args) {
        Mediator mediator = new StationManager();

        PassengerTrain passengerTrain = new PassengerTrain(mediator);
        FreightTrain freightTrain = new FreightTrain(mediator);

        passengerTrain.arrive();
        freightTrain.arrive();
        passengerTrain.depart();
    }
}

File: mediator/go/mediator_test.go
package mediator

import "testing"

func TestMediator(t *testing.T) {
	mediator := NewStationManger()

	passengerTrain := NewPassengerTrain(mediator)
	freightTrain := NewFreightTrain(mediator)

	passengerTrain.Arrive()
	freightTrain.Arrive()
	passengerTrain.Depart()
}
